src/main/java/org/spongepowered/common/registry/type/ArgumentRegistryModule.java

```diff
@@ -0,0 +1,87 @@
+package org.spongepowered.common.registry.type;
+
+import com.flowpowered.math.vector.Vector3d;
+import com.flowpowered.math.vector.Vector3i;
+import org.spongepowered.api.entity.EntityType;
+import org.spongepowered.api.entity.living.player.gamemode.GameMode;
+import org.spongepowered.api.text.selector.ArgumentHolder;
+import org.spongepowered.api.text.selector.ArgumentType;
+import org.spongepowered.api.text.selector.ArgumentTypes;
+import org.spongepowered.common.registry.RegisterCatalog;
+import org.spongepowered.common.registry.RegistrationDependency;
+import org.spongepowered.common.registry.RegistryModule;
+import org.spongepowered.common.registry.factory.SelectorFactoryModule;
+import org.spongepowered.common.text.selector.SpongeArgumentHolder;
+import org.spongepowered.common.text.selector.SpongeSelectorFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RegistrationDependency({SelectorFactoryModule.class, SelectorTypeRegistryModule.class})
+public class ArgumentRegistryModule implements RegistryModule {
+
+    @RegisterCatalog(ArgumentTypes.class)
+    private final Map<String, ArgumentHolder<?>> argumentTypeMap = new HashMap<>();
```

Comment:
The annotation is neat, but perhaps all that is needed is to parameterize RegistryModule and add a method `Map<String, T> getMap()`, it would be simpler